code/modules/clothing/under/accessories/holster.dm

```diff
@@ -171,3 +171,40 @@
 	name = "brown thigh holster"
 	icon_state = "holster_brown_thigh"
 
+// Custodial Holster
+// Only allows the pestgun and xenogun.
+/obj/item/clothing/accessory/holster/custodial/armpit/brown
+	name = "brown custodial armpit holster"
+	icon_state = "holster_brown"
+	var/list/allowedObjects = list(
+		/obj/item/gun/energy/mousegun,
+		/obj/item/gun/energy/mousegun/xenofauna,
+		/obj/item/gun/projectile/revolver/capgun,
+		/obj/item/toy/crossbow,
+		/obj/item/reagent_containers/food/snacks/grown/banana
+	)
+
+/obj/item/clothing/accessory/holster/custodial/armpit/brown/holster(var/obj/item/I, var/mob/living/user)
+	if(holstered && istype(user))
+		to_chat(user, "<span class='warning'>There is already \a [holstered] holstered here!</span>")
+		return
+
+	if(!(I.slot_flags & SLOT_HOLSTER))
+		to_chat(user, "<span class='warning'>[I] won't fit in [src]!</span>")
+		return
+
+	if(!is_type_in_list(I, allowedObjects))
+		to_chat(user, "<span class='warning'>[I] won't fit in [src]!</span>")
+		return
+
+	if(sound_in)
+		playsound(get_turf(src), sound_in, 50)
```

Comment:
A lot of unnecessary copypaste here. Basically, you only need to do the type_in_list check and then do ..().

Reply:
fixed!